### PR TITLE
workflow: add clippy and cargo-deny linters

### LINKED ISSUE #12: Umbrella issus for basic function
## Description
This is an umbrella issue to track some basic functions.

## CI/CD
- [x] Run `Cargo clippy -D warning` and `Cargo deny` in Github action  #22 @dierbei 
- [ ] Prepare for the physical machine: Postpone
- [x] #4 @flyflypeng 

## Functions
- [x] Privileged container in vmm-sandboxer : Supported
- [x] Resource usage stats for both container and sandbox : Supported
- [x] Time synchronization between Host and Guest VM @Burning1020 #50 
- [x] Limit VM process into cgroup filesystem @flyflypeng #51 
- [x] Sandbox files persist and recover @Vanient #40

### PR DESCRIPTION
**Pull request by dierbei**
adds a GitHub workflow that performs clippy linting and cargo-deny auditing on Rust code.

Fixes #12

### CODE REVIEW

**Comment by dierbei** [comment]
Clippy error fixed.

File: vmm/sandbox/src/main.rs [hunk 1]
```diff
@@ -108,7 +106,7 @@ async fn main() -> anyhow::Result<()> {
         .format_timestamp_micros()
         .init();
     #[cfg(feature = "qemu")]
-    let sandboxer: KuasarSandboxer<QemuVMFactory, QemuHooks> = {
+    let _sandboxer: KuasarSandboxer<QemuVMFactory, QemuHooks> = {
```
Comment (Burning1020) [inline vmm/sandbox/src/main.rs:109]: These sandboxers are used in different features, use`#[allow(dead_code)]`

File: vmm/sandbox/src/network/address.rs [hunk 2]
```diff
@@ -35,6 +35,7 @@ pub struct CniIPAddress {
     pub mask: String,
 }
 
+#[allow(clippy::from_over_into)]
```
Comment (Burning1020) [inline vmm/sandbox/src/network/address.rs:38]: We can implement `From`

File: vmm/task/src/container.rs [hunk 3]
```diff
@@ -205,7 +205,7 @@ impl KuasarFactory {
         // pivot_root could not work with initramfs
         match get_mount_type("/") {
             Ok(m_type) => {
-                if m_type == "rootfs".to_string() {
+                if m_type == *"rootfs" {
```
Comment (Burning1020) [inline vmm/task/src/container.rs:208]: Why '*' is needed?

File: vmm/sandbox/src/main.rs (hunk 1 above)
Comment (dierbei) [reply vmm/sandbox/src/main.rs]: Already added #[allow(dead_code)]

File: vmm/sandbox/src/network/address.rs (hunk 2 above)
Comment (dierbei) [reply vmm/sandbox/src/network/address.rs]: Already added implement `From`

File: vmm/sandbox/src/main.rs [hunk 4]
```diff
@@ -138,7 +137,6 @@ async fn main() -> anyhow::Result<()> {
                 if !Path::new(&dir_path).exists() {
                     tokio::fs::create_dir_all(&dir_path).await.unwrap();
                 }
-                dir = Some(dir_path);
```
[the remark this answers is not in the document]
Comment (dierbei) [reply vmm/sandbox/src/main.rs:141]: I saw it was unused and then I deleted it. I'll add it back later

**Review by Vanient — CHANGES_REQUESTED** [review]

File: vmm/sandbox/src/stratovirt/mod.rs [hunk 5]
```diff
@@ -179,16 +179,16 @@ impl VM for StratoVirtVM {
...
+                let virtio_net_device = VirtioNetDevice::new()
+                    .id(&tap_info.id)
+                    .name(&tap_info.name)
+                    .mac_address(&tap_info.mac_address)
+                    .transport(Transport::Pci)
+                    .fds(fd_ints)
+                    .vhost(false)
+                    .vhostfds(vec![])
+                    .bus(Some(DEFAULT_PCIE_BUS.to_string()))
+                    .build();
```
[the remark this answers is not in the document]
Comment (dierbei) [reply vmm/sandbox/src/stratovirt/mod.rs:191]: Yes, clippy recommends not having too many parameters.

So I try to reduce the argument of the function.

[the remark this answers is not in the document]
Comment (dierbei) [reply vmm/sandbox/src/stratovirt/mod.rs:191]: ref: https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

[the remark this answers is not in the document]
Comment (Vanient) [reply vmm/sandbox/src/stratovirt/mod.rs:191]: OK

**Review by Vanient — APPROVED** [review]

**Review by Vanient — APPROVED** [review]

**Review by Vanient — APPROVED** [review]

**Review by Burning1020 — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**